server/src/main/java/org/apache/iotdb/db/metadata/PartialPath.java

```diff
@@ -99,6 +105,41 @@ public PartialPath concatNode(String node) {
     return nodes;
   }
 
+  public int getNodeLength() {
+    return nodes.length;
+  }
+
+  public PartialPath alterPrefixPath(PartialPath prefixPath) {
+    String[] newNodes = Arrays.copyOf(nodes, Math.max(nodes.length, prefixPath.getNodeLength()));
+    System.arraycopy(prefixPath.getNodes(), 0, newNodes, 0, prefixPath.getNodeLength());
+    return new PartialPath(newNodes);
+  }
+
+  public void setPrefixPath(PartialPath prefixPath) {
+    this.nodes = Arrays.copyOf(nodes, Math.max(nodes.length, prefixPath.getNodeLength()));
+    System.arraycopy(prefixPath.getNodes(), 0, nodes, 0, prefixPath.getNodeLength());
+    fullPath = String.join(TsFileConstant.PATH_SEPARATOR, nodes);
+  }
+
+  public boolean matchFullPath(String rPath) throws IllegalPathException {
+    return matchFullPath(new PartialPath(rPath));
+  }
+
+  public boolean matchFullPath(PartialPath rPath) {
```

Comment:
add javadoc and give an example for each case

Reply:
Added